zbus/src/fdo.rs

```diff
@@ -702,8 +704,13 @@ macro_rules! gen_dbus_proxy {
             fn get_connection_unix_user(&self, bus_name: BusName<'_>) -> Result<u32>;
 
             /// Gets the unique ID of the bus.
+            #[deprecated(since = "3.14.0", note = "Use `get_guid` instead")]
             fn get_id(&self) -> Result<String>;
 
+            /// Gets the unique ID of the bus.
+            #[dbus_proxy(name = "GetId")]
+            fn get_guid(&self) -> Result<Guid>;
```

Comment:
I think I would prefer to stick with method name, even if it may break user code (this is a rather small change). Or perhaps gate the new API(s) with a feature such as "no-deprecated" ?

Reply:
> I think I would prefer to stick with the original method name, even if it may break user code (this is a rather small change).

If we had a way to know with fair certainty that this will only break things for a small number of users, I'd agree 100% but unfortunately we don't.

Having said that, this seems more like a fix. So maybe it's not a bad option. I'll add `Deref<Target = str>` and `From<Guid> for String` impls to make things easier.